### ISSUE TITLE
NPM package

### ISSUE THREAD

**Issue by jskorepa** [sammydre/ts-for-gir#1]
It would be awesome to be able to just `npx ts-for-gjs -m Atk-1.0` to get typescript definitions.

I can do it, if you'd like.

Also: the 1300 lines of code monolithic file is hard to follow - it would be great if it was multiple smaller files. Maybe even split out gir parsing and .d.ts writing into separate modules? I am also willing to do this and submit PRs.

I guess that this question is about what are your plans for this - do you want to continue working on it, or was it one-off project?

**Comment by sammydre** [comment]
The main answer to this is: yes, I continue to be interested and want to support this project. And yes, I'd be happy to receive patches.

Right now I'm mostly focused on looking for areas where the generation is incomplete or incorrect -- I've not found any for a while, which bodes well. npx support sounds good - I don't know what is required there, so would be interested in a PR. If you'd like to look at refactoring; then I'm open to that given a reasonable design.

**Comment by ImADrafter** [comment]
Just publishing the package on NPM will unlock npx support, since it is just an npm feature. I think this would be a huge improvement. Maybe this process could be automated on push by using Github actions with [npm-publish action](https://github.com/marketplace/actions/npm-publish) so a new version will be published automatically for every new commit

**Comment by andyrichardson** [comment]
Would be great to see this on npm - anything I can do to help?

If there's room for it, snapshot releases of the build output would also be dope (i.e. users run `npm i -D https://github.com/sammydre/ts-for-gjs/releases/download/[some-tag]/output.tgz` to get the latest types).

**Comment by JumpLink** [comment]
@sammydre There is a two years old package of ts-for-gir on npm: https://www.npmjs.com/package/ts-for-gir

Is this one from you?

**Comment by sammydre** [comment]
Nope, I didn’t put it there. Perhaps we can reach out to whomever is marked
on npm and ask about it.

**Comment by JumpLink** [comment]
@andyrichardson, @ImADrafter, @jskorepa Do any of you know who put the package on npm?

**Comment by ImADrafter** [comment]
`npm info` gives the last publisher:
```
dist-tags:
latest: 1.0.1

published a year ago by codejamninja <jam@codejam.ninja>
```

Who aparently, has published [over 250 packages](https://www.npmjs.com/~codejamninja) (surely with good intentions). But now on, I don't know how to proceed. @JumpLink, you can't publish because of this already existing package ?

**Comment by JumpLink** [comment]
@ImADrafter I assumed that I could not overwrite the package, but did not try it. But I know who codejamninja is and will contact him. Thanks a lot :)

@clayrisser You are codejamninja, right?

**Comment by JumpLink** [comment]
@sammydre I have tried to reach @clayrisser on different ways, unfortunately unsuccessfully. I have now reported the package to NPM as spam, let's see what I get here as a response. If NPM says they can't do anything about it, we should think about renaming the project so we can republish it on NPM.

**Comment by JumpLink** [comment]
@sammydre Okay I plan to publish the package under an new Name `@ts-for-gir/cli`, this way we can also split this project into multiple packages like `@ts-for-gir/generator`.

**Comment by JumpLink** [comment]
Okay the package is now published as `@ts-for-gir/cli`

See also #82
